cmake/clang-format.cmake

```diff
@@ -0,0 +1,33 @@
+find_program(PYTHON_EXECUTABLE python3 python)
+
+if(NOT PYTHON_EXECUTABLE)
+  message(WARNING "Python not found. Skipping lint and format checks.")
+else()
+  if(DEFINED ENV{IS_CI})
+    set(IS_CI $ENV{IS_CI})
+  else()
+    set(IS_CI OFF)
+  endif()
+
+  if(IS_CI)
```

Comment:
Can you also update one of the github workflows to add a IS_CI environment variable?